### PR TITLE
cp: fix cp -dR --no-preserve=links d c should have different inodes

### PR DESCRIPTION
**Pull request by tommady**
fix issue #5308

### CODE REVIEW

File: src/uu/cp/src/cp.rs [hunk 1]
```diff
@@ -931,23 +931,33 @@ impl Options {
         };
 
         // Parse attributes to preserve
-        let attributes = if let Some(attribute_strs) = matches.get_many::<String>(options::PRESERVE)
-        {
-            if attribute_strs.len() == 0 {
+        let mut attributes =
+            if let Some(attribute_strs) = matches.get_many::<String>(options::PRESERVE) {
+                if attribute_strs.len() == 0 {
```
Comment (sylvestre) [inline src/uu/cp/src/cp.rs:936]: is_empty doesn't work here ?

File: src/uu/cp/src/cp.rs [hunk 2]
```diff
@@ -931,23 +931,33 @@ impl Options {
...
+            } else if matches.get_flag(options::PRESERVE_DEFAULT_ATTRIBUTES) {
                 Attributes::DEFAULT
             } else {
-                Attributes::parse_iter(attribute_strs)?
+                Attributes::NONE
+            };
+
+        // handling no-preserve options and adjusting the attributes
+        if let Some(attribute_strs) = matches.get_many::<String>(options::NO_PRESERVE) {
+            if attribute_strs.len() > 0 {
```
Comment (sylvestre) [inline src/uu/cp/src/cp.rs:954]: same
! empty

**Comment by tommady** [comment]
Hi the is_empty method for the clap ValuesRef struct right now is a nightly-only experimental API.

So we could not using it right now.

Thanks.

**Comment by sylvestre** [comment]
looks great, thanks

**Pull request merged**